Firestore/Source/Core/FSTQuery.mm

```diff
@@ -77,15 +78,13 @@ + (instancetype)filterWithField:(const FieldPath &)field
                           value:(FSTFieldValue *)value {
   if ([value isEqual:[FSTNullValue nullValue]]) {
     if (op != FSTRelationFilterOperatorEqual) {
-      FSTThrowInvalidUsage(@"InvalidQueryException",
-                           @"Invalid Query. You can only perform equality comparisons on nil / "
-                            "NSNull.");
+      ThrowInvalidArgument(
+          "Invalid Query. You can only perform equality comparisons on nil / NSNull.");
     }
     return [[FSTNullFilter alloc] initWithField:field];
   } else if ([value isEqual:[FSTDoubleValue nanValue]]) {
     if (op != FSTRelationFilterOperatorEqual) {
-      FSTThrowInvalidUsage(@"InvalidQueryException",
-                           @"Invalid Query. You can only perform equality comparisons on NaN.");
+      ThrowInvalidArgument("Invalid Query. You can only perform equality comparisons on NaN.");
```

Comment:
Very optional, because it's not related to this PR: I find this wording a little ambiguous. I first parsed the sentence as _"only NaN supports equality comparisons"_ (which was very confusing), before I realized that the intended meaning is _"NaN supports only equality comparisons"_ (I had to glance at the code to figure it out). Maybe:
* `NaN supports only equality comparisons`
* `NaN doesn't support any kind of comparisons except equality`

Same applies to the message above.

Reply:
OK. I chose the former and applied it to both.